deep-learning/src/main/scala/com/microsoft/azure/synapse/ml/onnx/ONNXHub.scala

```diff
@@ -0,0 +1,229 @@
+
+    // Filter by model name first.
+    val matchingModels = model.map(m => manifest.filter(_.model.toLowerCase() == m.toLowerCase)).getOrElse(manifest)
+
+    // Filter by tags
+    if (tags.isEmpty) {
+      matchingModels
+    } else {
+      val canonicalTags = tags.get.map(_.toLowerCase).toSet
+      matchingModels
+        .filter(_.metadata.tags.isDefined)
+        .filter(_.metadata.tags.get.map(_.toLowerCase).toSet.intersect(canonicalTags).nonEmpty)
+    }
+  }
+
+  def getModelInfo(model: String, repo: String = ONNXHub.DefaultRepo, opset: Option[Int] = None): ONNXModelInfo = {
+    val matchingModels = listModels(repo, Some(model))
+    if (matchingModels.isEmpty)
+      throw new IllegalArgumentException(s"No models found with name $model")
+
+    val selectedModels = opset.map(ov => matchingModels.filter(_.opsetVersion == ov))
+      .getOrElse(matchingModels.sortBy(-_.opsetVersion))
+
+    if (selectedModels.isEmpty) {
+      val validOpsets = matchingModels.map(_.opsetVersion)
+      throw new IllegalArgumentException(s"$model has no version with opset $opset. Valid opsets: $validOpsets")
+    }
+    selectedModels.head
+  }
+
+  //noinspection ScalaStyle
+  private def downloadModel(url: URL, path: Path, fs: FileSystem): Unit = {
+    FaultToleranceUtils.retryWithTimeout(3, Duration.apply(10, "minutes")) {
+      val urlCon = url.openConnection()
+      urlCon.setConnectTimeout(ONNXHub.ConnectTimeout)
+      urlCon.setReadTimeout(ONNXHub.ReadTimeout)
+      using(new BufferedInputStream(urlCon.getInputStream)) { is =>
+          using(fs.create(path)) { os =>
+          HUtils.copyBytes(is, os, SparkContext.getOrCreate().hadoopConfiguration)
+        }
+      }
+    }
+  }
+
+  def load(model: String,
+           repo: String = ONNXHub.DefaultRepo,
+           opset: Option[Int] = None,
+           forceReload: Boolean = false,
+           silent: Boolean = false): Array[Byte] = {
+    val selectedModel = getModelInfo(model, repo, opset)
+    val modelPathArr = selectedModel.modelPath.split("/".toCharArray)
+    val localModelDirs: Seq[String] = modelPathArr.dropRight(1) ++
+    Seq(selectedModel.metadata.modelSha.map(sha => s"${sha}_${modelPathArr.last}").getOrElse(modelPathArr.last))
+
+    val localModelPath = localModelDirs.foldLeft(getDir) { case (path, folder) => new Path(path, folder) }
+    val fs = localModelPath.getFileSystem(SparkContext.getOrCreate().hadoopConfiguration)
+
+    if (forceReload || !fs.exists(localModelPath)) {
+      if (!verifyRepoRef(repo) && !silent) {
+        val message = s"""The model repo specification \"$repo\"
+           | is not trusted and may contain security vulnerabilities.
+           |  Only continue if you trust this repo.""".stripMargin
+        System.err.println(message)
+        println("Continue?[y/n]")
+        if (scala.io.StdIn.readLine().toLowerCase != "y") {
+          throw new InterruptedException("User has cancelled model downloading due to security warning")
+        }
```

Comment:
how would you suggest this interacts with the `silent ` argument?  (Mark wrote this)  I presume this scenario is what he added the silent parameter for?

Reply:
Good question. It's up to us to define the behavior. What do you think of the following?
When the repo is not verified to be safe:
1. if !allowUnsafe: throw error.
2. Otherwise if silent: load the model without any warning.
3. Otherwise, load the model but print the warning first.